calculadora de imc/js/index.js

```diff
@@ -3,59 +3,54 @@ const peso = document.querySelector('#peso');
 const altura = document.querySelector('#altura');
 
 
-botaoEnviar.addEventListener('click', ()=>{
-    const resultadoVerificacao = verificarEntradas();
+botaoEnviar.addEventListener('click', () => {
+    const altura = Number(altura.value)
+    const peso = Number(peso.value)
 
-    if(resultadoVerificacao !== true){
-        setarElemento('p', 'resultado-ruim', resultadoVerificacao);
-    }else if(resultadoVerificacao === true){
-        setarElemento('p', 'resultado-bom', resulImc(Number(peso.value), Number(altura.value)))
+    const error = validateImcInputs(altura, peso);
+
+    if (error) {
+        setResult(false, error);
+    } else {
+        setResult(true, getImcMessage(peso, altura))
     }
 });
 
-function verificarEntradas() {
-    let res;
-
-    if(!Number(altura.value) && !Number(peso.value)){
-        res = "Altura e Peso inválidos";
-    }else if(!Number(altura.value)){
-        res = "Altura inválida";
-    }else if(!Number(peso.value)){
-        res = "Peso inválido";
+function validateImcInputs(altura, peso) {
+    if(!altura && !peso) {
+        return "Altura e Peso inválidos";
+    } else if (!altura) {
+        return "Altura inválida";
+    } else if (!peso){
+        return "Peso inválido";
     }
-    else res = true;
-
-    return res;
+    else return undefined;
 }
 
-function setarElemento(elemento, classe, conteudo) {
-    const localResultado = document.querySelector('#resultado');
-    let element = document.createElement(elemento);
+function setResult(success, message) {
```

Comment:
Como essa função é bem específica do nosso elemento  #resultado , não precisamos de parametros generalizados como o elemento e classe (pois já sabemos exatamente quais os valores vão ser, e se for passado outros valores pode dar um erro)